Week-8/Ex8_2/rotateLinkedList.js

```diff
@@ -0,0 +1,80 @@
+class Node {
+  constructor(data, next = null) {
+    this.data = data;
+    this.next = next;
+  }
+}
+
+class LinkedList {
+  constructor() {
+    this.head = null;
+    this.size = 0;
+  }
+  // Insert first node
+  insertFirst(data) {
+    this.head = new Node(data, this.head);
+    this.size++;
+  }
+  // Insert at last
+  insertLast(data) {
+    let current = this.head;
+    let node = new Node(data);
+    if (this.head == null) {
+      this.head = node;
+    } else {
+      while (current.next) {
+        current = current.next;
+      }
+      current.next = node;
+    }
+    this.size++;
+  }
+  // Remove at index
+  removeAt(index) {
+    let current = this.head,
+      previous,
+      count = 0;
+    if (index === 0) {
+      this.head = current.next;
+    } else if (index > 0 && index < this.size) {
+      while (count < index) {
+        previous = current;
+        current = current.next;
+        count++;
+      }
+      previous.next = current.next;
+      current.next = null;
+    }
+    this.size--;
+  }
+
+  // Print list
+  printListData() {
+    let current = this.head,
+      output = "";
+    while (current) {
+      output += current.data + " -> ";
+      current = current.next;
+    }
+    console.log(output);
+  }
+
+  rotateList(k) {
+    let current;
+    for (let i = 1; i <= k; i++) {
```

Comment:
Quick Tip: What if the length of the linked list is N and we have to rotate it N times?
This means N = K. In this case there will be no change in the before - after of the linked list.
In such case instead of running the loop till K. We should remove all the unnecessary loops by K = K % N.
For example: N = 10, K =. 3, the loop will iterate 3 times.
Example 2: N = 10,  K = 33, the loop will iterate 33 times, but will give the same result as of K = 3. So we should K = K % N, K = 33 % 10. K = 3